pkl-commons-cli/src/main/kotlin/org/pkl/commons/cli/CliBaseOptions.kt

```diff
@@ -167,4 +168,26 @@ data class CliBaseOptions(
     projectDir?.resolve(ProjectDependenciesManager.PKL_PROJECT_FILENAME)
       ?: normalizedWorkingDir.getProjectFile(rootDir)
   }
+
+  /** [caCertificates] after normalization. */
+  val normalizedCaCertificates: List<Path> = caCertificates.map(normalizedWorkingDir::resolve)
+
+  /**
+   * The HTTP client shared between CLI commands created with this [CliBaseOptions] instance.
+   *
+   * To release the resources held by the HTTP client in a timely manner, call its `close()` method.
+   */
+  val httpClient: HttpClient by lazy {
+    val builder = HttpClient.builder()
+    if (normalizedCaCertificates.isEmpty()) {
+      builder.addDefaultCliCertificates()
+    } else {
+      for (file in normalizedCaCertificates) builder.addCertificates(file)
+    }
+    // Lazy building significantly reduces execution time of commands that do minimal work.
+    // However, it means that HTTP client initialization errors won't surface until an HTTP request
+    // is
+    // made. A middleground would be to only build lazily if built-in certificates are used.
+    builder.buildLazily()
```

Comment:
> Move built-in CA certs back to pkl-commons

They never were in pkl-commons. Details elsewhere.

> Let's not touch the executor API

Further discussed elsewhere.

> EvaluatorBuilder.preconfigured() should trust the JVM's certs

I think the best I can do is to use `SSLContext.getDefault()`, which is more or less what you're asking for.

> Also: what was the thinking around introducing org.pkl.core.http.HttpClient?

At first I was using `java.net.http.HttpClient` directly. But baking this interface into Pkl APIs didn't feel right. Having our own abstraction has clear benefits:

* It puts us in control: We can remove what is irrelevant to Pkl, and add or change what is beneficial.
* A smaller interface is easier to implement and doesn't require throwing `UnsupportedOperationException` for all the methods that a particular implementation may not be able or willing to support. There are already 4 implementations, and chances are there will be more.
* For the same reason, it will be easier to support alternative implementations (Apache, Netty, etc.) if the need ever arises. While `java.net.http.HttpClient` is a clear step up from `HttpURLConnection`, it will always remain less capable than third-party libraries. For example, Netty supports a safer TLS implementation.

> But, it also puts the burden on us to expose all the settings someone might want, that are already in java.net.http.HttpClient.Builder.
> Also, maybe users already have their own java.net.HttpClient that they want to use and share with the rest of their application?

We could easily support this with something like an `HttpClient from(java.net.http.HttpClient)` factory method.

I didn't want to go so far as to have our own `HttpRequest`, `HttpResponse`, etc. Having "just" our own `HttpClient` felt like the sweet spot for Pkl.

Reply:
> `apply` is a nice way to reduce a bit of boilerplate.

Changed to `with(HttpClient.builder()) {...}`.